modules/security/src/main/java/io/oasp/module/security/common/base/accesscontrol/AbstractAccessControlProvider.java

```diff
@@ -67,18 +65,25 @@ protected void initialize(AccessControlSchema config) {
    * graph}.
    *
    * @param group is the {@link AccessControlGroup} to check.
-   * @param nodeCycle the {@link NodeCycle} used to detect cycles.
+   * @param groupList the {@link List} of visited {@link AccessControlGroup}s used to detect cycles.
    */
-  protected void checkForCyclicDependencies(AccessControlGroup group, NodeCycle<AccessControlGroup> nodeCycle) {
+  protected void checkForCyclicDependencies(AccessControlGroup group, List<AccessControlGroup> groupList) {
```

Comment:
The refactoring is fine, but is there any reason, why we pass and return values by parameter reference?
Just from reading, you could also remove the second parameter and delcare it as a return value.

Reply:
thanks for your feedback.
IMHO your observation is missing the fact that the method is recursive and additionally modifying the state of the List after the recursive call. This would not work without the parameter.

Reply:
Oh missed that, thanks.